pyqtgraph/graphicsItems/ImageItem.py

```diff
@@ -640,7 +640,7 @@ def _try_rescale_float(self, image, levels, lut):
 
         minVal, maxVal = levels
         if minVal == maxVal:
-            maxVal = xp.nextafter(maxVal, 2*maxVal)
+            maxVal = xp.nextafter(maxVal, 2*maxVal) if maxVal != 0 else numpy.finfo(maxVal).eps
         rng = maxVal - minVal
         rng = 1 if rng == 0 else rng
```

Comment:
This (pre-existing) line seems redundant if the code above works as intended. If it is indeed still needed, then there's a kind of problem that gets past the check `minVal == maxVal`... Possibly something originating from mixed or non-float datatypes?

Is the real thing we want to avoid ```float(maxval - minval) == 0``` ? Then the most robust path might be to *directly* check that. Something like
```
rng = float(maxVal - minVal)
if rng == 0.:
   maxVal = math.nextafter( minVal, math.inf )
   rng = float(maxVal - minVal)
```